base-isa.md

```diff
@@ -22,48 +23,45 @@ The highest two bits of the first byte are a format marker:
-| `01 SS CCCC`  | `RRR IIIII`  | when `SS != 01`, reserved for extensions |
+| `0? SS CCCC`  | `???? ????`  | when `SS != 01`, reserved for extensions |
 | `10 0 D CCCC` | `DDDDDDDD`   | (conditional) relative jump instruction  |
 | `10 1 ?????`  | `?????????`  | reserved for extension                   |
 | `11 ??????`   | `?????????`  | reserved for extensions                  |
 
-| Symbol | Meaning                      |
-|--------|------------------------------|
-| C      | opcode bit                   |
-| R      | register id                  |
-| I      | immediate                    |
-| D      | displacement                 |
-| S      | size (res.d)                 |
-| ?      | reserved for unknown purpose |
+| Symbol | Meaning         |
+|--------|-----------------|
+| C      | Opcode          |
+| R      | Register Id     |
+| I      | Immediate       |
+| D      | Displacement    |
+| S      | Operation Size  |
+| M      | Memory Mode     |
+| ?      | Arbitrary Value |
 
-### Execution of Reserved Instructions
-
-Execution of a reserved instruction is undefined behavior. A future extension will change this to "the CPU must trigger a specific interrupt."
+### Execution of Illegal and Reserved Instructions
 
+Execution of an _illegal_ or _reserved_ instruction is _unspecified_ behavior. A future extension will change this to require an interrupt to occur.
 
 ## Computation Instructions
 
-Both computation formats share a lot of similarities. For both the first byte has the structure `0x SS CCCC`.
+Both computation formats share a lot of similarities. For both, the first byte has the structure `0x SS CCCC`.
 
-- `SS` is a size marker and reserved for extensions. For the base instructions this should always be `01`
-- `CCCC` is a 4bit opcode deciding which operation to execute. This can be an ALU or a memory load/store instruction
+- `SS` is a size marker and reserved for extensions. For the base instructions this _must_ always be `01`
+- `CCCC` is a 4 bit opcode deciding which operation to execute.
 
 ### 2 Register Computation
 
-The second byte has the format `AAA BBB 00` where `AAA` and `BBB` are references to registers. `AAA` is the destination register and left source register. `BBB` is the right source register. The only exception is the RSUB instruction where `AAA` and `BBB` are swapped for the purpose of being source registers.
+The second byte has the format `AAA BBB MM` where `AAA` and `BBB` are references to registers. `AAA` is the destination register and left source register. `BBB` is the right source register. There are 2 exceptions to this. The first is the RSUB instruction where `AAA` and `BBB` are swapped for the purpose of being source registers. The second exception is that the `B` argument specifies which memory address or control register to write to. `MM` is the memory mode which will be defined in an extension and is only defined for `00` in the base isa.
```

Comment:
```suggestion
The second byte has the format `AAA BBB 00` where `AAA` and `BBB` are references to registers. `AAA` is the destination register and left source register. `BBB` is the right source register. The only exception is the RSUB instruction where `AAA` and `BBB` are swapped for the purpose of being source registers.
```

(Suggesting revert) No need to complicate the specification of base by explaining how it interacts with an extension that it doesn't require...

References to "memory" only apply to non-base instructions (`MM != 00`) which aren't even defined here. For all a reader knows, those formats could be _completely_ different (and, in a sense, they are, since some data is shifted to a third byte in many cases). `BBB` never specifies a control register in this format, because `READCR` and `WRITECR` opcodes do not have 2r forms.

Reply:
Mega had suggested that we explicitly mention how the `BBB` acts as a memory address/control register ID since others had gotten confused about it.

Reply:
`STORE` also uses ``BBB`` as the destionation and does have a 2 register form. This entire section is actually also not only applicable to the 2-reg addressing, but also for the reg-imm addressing. Maybe it shouldn't be called `BBB` and `AAA` here then.

Reply:
This section is right under the "2 register computation" header. Are you proposing putting it somewhere else?

Reply:
I moved the exceptions to there own section. How's it look?